pkg/xds/context/mesh_context_builder.go

```diff
@@ -298,12 +272,60 @@ func (m *meshContextBuilder) BuildBaseMeshContextIfChanged(ctx context.Context,
 		return latest, nil
 	}
 	return &BaseMeshContext{
-		hash:        newHash,
-		Mesh:        mesh,
-		ResourceMap: rmap,
+		hash:             newHash,
+		Mesh:             mesh,
+		ResourceMap:      rmap,
+		DestinationIndex: buildDestinationsIndex(rmap),
 	}, nil
 }
 
+func buildDestinationsIndex(resourceMap ResourceMap) *DestinationIndex {
```

Comment:
Feels like this should be NewDestinationIndex inside the destination_index.go

Reply:
moved